plugin/pkg/scheduler/scheduler.go

```diff
@@ -176,6 +190,43 @@ func (sched *Scheduler) schedule(pod *v1.Pod) (string, error) {
 	return host, err
 }
 
+func (sched *Scheduler) preempt(preemptor *v1.Pod, scheduleErr error) (string, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.PodPriority) {
+		glog.V(3).Infof("Pod priority feature is not enabled. No preemption is performed.")
+		return "", nil
+	}
+	preemptor, err := sched.config.PodPreemptor.GetUpdatedPod(preemptor)
+	if err != nil {
+		glog.Errorf("Error getting the updated preemptor pod object: %v", err)
+		return "", err
+	}
+	node, victims, err := sched.config.Algorithm.Preempt(preemptor, sched.config.NodeLister, scheduleErr)
+	if err != nil {
+		glog.Errorf("Error preempting victims to make room for %v/%v.", preemptor.Namespace, preemptor.Name)
+	}
+	if node == nil {
+		return "", err
+	}
+	glog.Infof("Preempting %d pod(s) on node %v to make room for %v/%v.", len(victims), node.Name, preemptor.Namespace, preemptor.Name)
+	annotations := map[string]string{core.NominatedNodeAnnotationKey: node.Name}
+	err = sched.config.PodPreemptor.UpdatePodAnnotations(preemptor, annotations)
+	if err != nil {
+		glog.Errorf("Error in preemption process. Cannot update pod %v annotations: %v", preemptor.Name, err)
+		return "", err
+	}
+	var victimNames bytes.Buffer
+	for _, victim := range victims {
+		if err := sched.config.PodPreemptor.DeletePod(victim); err != nil {
+			glog.Errorf("Error preempting pod %v: %v", victim.Name, err)
+			return "", err
+		}
+		victimNames.WriteString(fmt.Sprintf("%v/%v, ", victim.Namespace, victim.Name))
+		sched.config.Recorder.Eventf(victim, v1.EventTypeNormal, "GotPreempted", "by %v/%v on node %v", preemptor.Namespace, preemptor.Name, node.Name)
+	}
+	sched.config.Recorder.Eventf(preemptor, v1.EventTypeNormal, "Preempted", "pods %v on node %v", victimNames.String(), node.Name)
```

Comment:
I don't think this is a good use of even - this is basically duplication of those events above.

With the new event API where we will have clearly subject and object of an action and it will be extremely easy to find victims anyway. 

But no matter of that, please remove this event as a duplication of those above

Reply:
David asked me to add this and I think it makes sense to have them. The victims are deleted soon after preemption, but the preemptor stays. So, it would be more helpful to have these events for the pod that lives longer.

Reply:
I don't agree with that suggestion.

The fact that you are deleting a Pod doesn't mean that events associated with that pod are also deleted - they aren't.
Events are deleted in three situations:
- when the whole namespace is deleted
- if you explicitly delete them (but noone is doing that)
- when TTL elapses

With the whole effort of cleaning up events, I would really like to remove this.

@davidopp - I'm happy to discuss in person, I'm really convinced that is not a good idea.

Reply:
Sorry, I was confused. Somehow I thought events were deleted when the associated object was deleted. Thanks for the correction. @bsalamat I think deleting this makes sense (going back to the original version you had :) )

Reply:
Thanks, Wojtek! Removed it.